sql/mysqld.cc

```diff
@@ -3287,7 +3287,11 @@ pthread_handler_t signal_hand(void *arg __attribute__((unused)))
       }
       break;
     case SIGHUP:
+#ifdef SI_KERNEL
       if (!abort_loop && origin != SI_KERNEL)
+#else
+      if (!abort_loop)
```

Comment:
For illumos, you probably want 
`if (!abort_loop && origin <= SI_USER)`
See https://github.com/omniosorg/illumos-omnios/blob/master/usr/src/uts/common/sys/siginfo.h#L124
- codes > 0 are from the kernel.